diff -ur Random123-1.09.orig/include/Random123/features/gccfeatures.h Random123-1.09/include/Random123/features/gccfeatures.h
--- Random123-1.09.orig/include/Random123/features/gccfeatures.h	2018-12-28 18:15:37.119315452 +0000
+++ Random123-1.09/include/Random123/features/gccfeatures.h	2018-12-28 18:16:08.363157300 +0000
@@ -34,7 +34,7 @@
 
 #define R123_GNUC_VERSION (__GNUC__*10000 + __GNUC_MINOR__*100 + __GNUC_PATCHLEVEL__)
 
-#if !defined(__x86_64__) && !defined(__i386__) && !defined(__powerpc__)
+#if !defined(__x86_64__) && !defined(__i386__) && !defined(__powerpc__)  && !defined(__aarch64__)
 #  error "This code has only been tested on x86 and powerpc platforms."
 #include <including_a_nonexistent_file_will_stop_some_compilers_from_continuing_with_a_hopeless_task>
 { /* maybe an unbalanced brace will terminate the compilation */
